File: src/main/java/com/freya02/botcommands/api/core/DefaultEmbedFooterIconSupplier.java
package com.freya02.botcommands.api.core;

import com.freya02.botcommands.api.commands.prefixed.BaseCommandEvent;
import com.freya02.botcommands.api.core.config.BServiceConfigBuilder;
import com.freya02.botcommands.api.core.service.annotations.BService;
import com.freya02.botcommands.api.core.service.annotations.InterfacedService;
import org.jetbrains.annotations.Nullable;

import java.io.InputStream;

/**
 * Interface for embed footer icons requested by {@link BaseCommandEvent#getDefaultIconStream()}.
 *
 * <p>
 * <b>Usage</b>: Register your instance as a service with {@link BService}
 * or {@link BServiceConfigBuilder#getServiceAnnotations() any annotation that enables your class for dependency injection}.
 *
 * @see InterfacedService @InterfacedService
 */
@InterfacedService(acceptMultiple = false)
public interface DefaultEmbedFooterIconSupplier {
	class Default implements DefaultEmbedFooterIconSupplier {
		@Nullable
		@Override
		public InputStream get() {
			return null;
		}
	}

	@Nullable
	InputStream get();
}

File: BotCommands-core/src/main/kotlin/io/github/freya022/botcommands/api/core/DefaultEmbedFooterIconSupplier.kt
package io.github.freya022.botcommands.api.core

import io.github.freya022.botcommands.api.commands.text.BaseCommandEvent
import io.github.freya022.botcommands.api.core.service.annotations.BService
import io.github.freya022.botcommands.api.core.service.annotations.InterfacedService
import java.io.InputStream

/**
 * Interface for embed footer icons requested by [BaseCommandEvent.getDefaultIconStream].
 *
 * Returns `null` by default.
 *
 * **Usage**: Register your instance as a service with [BService].
 *
 * @see InterfacedService @InterfacedService
 */
@InterfacedService(acceptMultiple = false)
fun interface DefaultEmbedFooterIconSupplier {

    fun get(): InputStream?
}
